docs/src/dev-guide/coding-guidelines.md

```
@@ -53,6 +53,190 @@ To avoid including a state variable in a React Hook's dependency array, you can
+    },
+}));
+```
+
+## Feature-based slicing
+
+When a Zustand store grows too large, split it into slices based on features such as functional
+areas.
+
+:::{warning}
+Follow the principle of separation of concerns:
+- Do: Slice by feature. e.g., query configuration, query results, or query controller.
```

markdownlint-cli2 (0.17.2)

Unordered list style
Expected: asterisk; Actual: dash

(MD004, ul-style)